### PR TITLE
Build on ubuntu 18.04 (bionic)

### PR DESCRIPTION
**Pull request by shaitan**

### CODE REVIEW

File: library/protocol.cpp [hunk 1]
```diff
@@ -5,513 +5,604 @@
...
+		if (o.type!=msgpack::type::ARRAY || o.via.array.size != 2) {
+			throw msgpack::type_error();
+		}
 
-inline dnet_read_request &operator >>(msgpack::object o, dnet_read_request &v) {
-	if (o.type != msgpack::type::ARRAY || o.via.array.size < 4) {
-		throw msgpack::type_error();
+		auto p = o.via.array.ptr;
+		p[0].convert(v.tsec);
+		p[1].convert(v.tnsec);
```
Comment (sabramkin) [inline library/protocol.cpp]: JFMI: Before commit here was `&v.tnsec` - what a reason to change?

File: library/protocol.cpp [hunk 2]
```diff
@@ -5,513 +5,604 @@
...
-
-template <typename Stream>
-inline msgpack::packer<Stream> &operator <<(msgpack::packer<Stream> &o, const dnet_time &v) {
-	o.pack_array(2);
-	o.pack_fix_uint64(v.tsec);
-	o.pack_fix_uint64(v.tnsec);
-	return o;
-}
+template<>
+struct convert<dnet_time> {
```
Comment (sabramkin) [inline library/protocol.cpp]: JFMI: Am I understand correctly: it's a convient way to create custom convertors, and this opportunity is introduced in `v2.1.5-1`? In previous code used stream pattern, but now used conversion "method".

File: library/protocol.cpp (hunk 1 above)
Comment (shaitan) [reply library/protocol.cpp]: msgpack API changes, now it accepts references instead of pointers.

File: library/protocol.cpp (hunk 2 above)
Comment (shaitan) [reply library/protocol.cpp]: Yes, but it was introduced in v1.0 or v.1.1 (https://github.com/msgpack/msgpack-c/wiki/v1_1_cpp_adaptor#non-intrusive-approach).

**Pull request closed without merging**